### DISCUSSION TITLE
How to specify GPU for single machine multi-gpus training

### DISCUSSION THREAD

**Discussion by zhaijianyang** [open-mmlab/mmaction2#691] [Q&A] [answered]
How to specify GPU for single machine multi-gpus training?

**Comment by irvingzhang0512** [comment]
You could use `--gpu-ids`

https://github.com/open-mmlab/mmaction2/blob/ee34d952e792fd1adea2c2e397b29faff68eaec9/tools/train.py#L37-L42

But I prefer setting `CUDA_VISIBLE_DEVICES`

```shell
CUDA_VISIBLE_DEVICES=1 python tools/train.py xxx
```

**Comment by congee524** [comment, marked as answer]
Hi, you can also refer to [launch-multiple-jobs-on-a-single-machine](https://github.com/open-mmlab/mmaction2/blob/master/docs/getting_started.md#launch-multiple-jobs-on-a-single-machine)

**Comment by zhaijianyang** [comment]
> You could use `--gpu-ids`
> 
> https://github.com/open-mmlab/mmaction2/blob/ee34d952e792fd1adea2c2e397b29faff68eaec9/tools/train.py#L37-L42
> 
> But I prefer setting `CUDA_VISIBLE_DEVICES`
> 
> ```shell
> CUDA_VISIBLE_DEVICES=1 python tools/train.py xxx
> ```

ok,thank you
